sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -162,6 +162,8 @@ trait FileFormat {
    * By default all data types are supported.
    */
   def supportDataType(dataType: DataType): Boolean = true
+
+  def checkFieldNames(names: Seq[String]): Unit = {}
```

Comment:
Can we add some document for this API? We can probably require the exception type as `AnalysisException`